src/main/java/de/dhbw/mh/slang/codegen/JavaBytecodeGenerator.java

```diff
@@ -88,16 +90,52 @@ public String visitPost( AstBinaryOperation node, String lhs, String rhs ){
 				throw new RuntimeException( "not yet implemented" );
 			}
 			case LESS_THAN:{
-				throw new RuntimeException( "not yet implemented" );
+				String result = "";
+				result += lhs;
+				result += rhs;
+				result += "if_icmplt #true_lt" + counter_greater_less_equals + System.lineSeparator();
+				result += "iconst_0" + System.lineSeparator();
+				result += "goto #end_lt" + counter_greater_less_equals + System.lineSeparator();
+				result += "#true_lt" + counter_greater_less_equals + System.lineSeparator();
+				result += "iconst_1" + System.lineSeparator();
+				result += "#end_lt" + counter_greater_less_equals + System.lineSeparator();
+				return result;
 			}
 			case GREATER_OR_EQUAL:{
-				throw new RuntimeException( "not yet implemented" );
+				String result = "";
+				result += lhs;
+				result += rhs;
+				result += "if_icmpge #true_ge" + counter_greater_less_equals + System.lineSeparator();
+				result += "iconst_0" + System.lineSeparator();
+				result += "goto #end_ge" + counter_greater_less_equals + System.lineSeparator();
+				result += "#true_ge" + counter_greater_less_equals + System.lineSeparator();
+				result += "iconst_1" + System.lineSeparator();
+				result += "#end_ge" + counter_greater_less_equals + System.lineSeparator();
+				return result;
 			}
 			case GREATER_THAN:{
-				throw new RuntimeException( "not yet implemented" );
+				String result = "";
+				result += lhs;
+				result += rhs;
+				result += "if_icmpgt #true_gt" + counter_greater_less_equals + System.lineSeparator();
+				result += "iconst_0" + System.lineSeparator();
+				result += "goto #end_gt" + counter_greater_less_equals + System.lineSeparator();
+				result += "#true_gt" + counter_greater_less_equals + System.lineSeparator();
+				result += "iconst_1" + System.lineSeparator();
+				result += "#end_gt" + counter_greater_less_equals + System.lineSeparator();
```

Comment:
Although the test requires a different solution, yours is equally valid. I am also willing to accept yours and rewrite your individual test case. But please stick to the naming convention of the other labels: # + ID + suffix. You can use the `StringBuilder` and/or `String.format` if this becomes too confusing for you.

It should also be noted that the inversion of the original operator does not occur without reason. Normally, a comparison is associated with the assumption that it is true in most cases and the conditional jump is not executed. This may give a slight performance advantage in some circumstances. Apart from that, it is also more logical to jump when a comparison fails than when it applies.

Still, both solutions are technically identical and I'm happy to adjust the test if you insist on your solution.